### PR TITLE
Add Traverse[Try]

### LINKED ISSUE #1081: Add Traverse[Try] instance
Follow up to https://github.com/typelevel/cats/pull/1059#issuecomment-222554220

### PR DESCRIPTION
**Pull request by ceedubs**
Resolves #1081

### CODE REVIEW

**Comment by non** [comment]
👍

**Comment by ceedubs** [comment]
@johnynek would you be interested in reviewing this one, since you recently submitted #1059?

File: core/src/main/scala/cats/std/try.scala [hunk 1]
```diff
@@ -78,6 +97,14 @@ trait TryInstances extends TryInstances1 {
     }
 }
 
+private[std] object TryInstances {
+  /**
+   * A `Failure` can be statically typed as `Try[A]` for all `A`, because it
+   * does not actually contain an `A` value (as `Success[A]` does).
+   */
+  def castFailure[A](f: Failure[_]): Try[A] = f.asInstanceOf[Try[A]]
```
Comment (johnynek) [inline core/src/main/scala/cats/std/try.scala]: worth it to put an `@inline` here? I don't actually know how effective that is, but it seems like a candidate.

**Comment by johnynek** [comment]
👍 modulo the minor comment on `@inline`

File: core/src/main/scala/cats/std/try.scala (hunk 1 above)
Comment (ceedubs) [inline core/src/main/scala/cats/std/try.scala]: Good suggestion. That seems reasonable to me.

**Comment by ceedubs** [comment]
I've updated with `@inline`

**Comment by johnynek** [comment]
:+1:

**Comment by adelbertc** [comment]
:+1:

**Pull request merged**